### ISSUE TITLE
How do I change resolution on CaptureDevice?

### ISSUE THREAD

**Issue by patrickhener** [hybridgroup/gocv#714]
## Description
I want to snap a pic and I have this FullHD camera. But when I use saveimage example it will just record an image with the resolution of 640x480.

## Steps to Reproduce
Use saveimage from ./cmd/saveimage/main.go with fullhd camera (1920x1080 FullHD Camera)

1. go run ./cmd/saveimage/main.go

## Your Environment

* Operating System and version: Linux hummingbird 5.7.8-arch1-1 #1 SMP PREEMPT Thu, 09 Jul 2020 16:34:01 +0000 x86_64 GNU/Linux
* OpenCV version used: 4.3.0
* How did you install OpenCV?: Arch User Repository
* GoCV version used: 0.23.0
* Go version: 1.14.5 linux/amd64

Basically I am looking for this settings: [https://answers.opencv.org/question/34461/how-to-set-camera-resolution-webcam-with-opencv/](https://answers.opencv.org/question/34461/how-to-set-camera-resolution-webcam-with-opencv/) in gocv.

**Comment by inikotoran** [comment]
have you tried this?
```
webcam.Set(gocv.VideoCaptureFrameWidth, 1920.0)
webcam.Set(gocv.VideoCaptureFrameHeight, 1080.0)
```

**Comment by patrickhener** [comment]
I tested with the example program "capwindow". Without the settings it will give me an output window of resolution: 640x480 and will be working. When setting the options you suggested I will get an error:

```
❯ go run main.go 4
[ WARN:0] global /build/opencv/src/opencv-4.4.0/modules/videoio/src/cap_gstreamer.cpp (935) open OpenCV | GStreamer warning: Cannot query video position: status=0, value=-1, duration=-1
[ WARN:0] global /build/opencv/src/opencv-4.4.0/modules/videoio/src/cap_gstreamer.cpp (1760) handleMessage OpenCV | GStreamer warning: Embedded video playback halted; module v4l2src0 reported: Internal data stream error.
[ WARN:0] global /build/opencv/src/opencv-4.4.0/modules/videoio/src/cap_gstreamer.cpp (515) startPipeline OpenCV | GStreamer warning: unable to start pipeline
[ WARN:0] global /build/opencv/src/opencv-4.4.0/modules/videoio/src/cap_gstreamer.cpp (1057) setProperty OpenCV | GStreamer warning: no pipeline
Start reading device: 4
Device closed: 4
[ WARN:0] global /build/opencv/src/opencv-4.4.0/modules/videoio/src/cap_gstreamer.cpp (480) isPipelinePlaying OpenCV | GStreamer warning: GStreamer: pipeline have not been created
```
Same applies to the mentioned test program "saveimage".

**Comment by deadprogram** [comment]
Keep in mind that you can only capture at resolution(s) supported by that camera, just in case that is a factor.

**Comment by patrickhener** [comment]
@deadprogram Thanks for the hint but it is a 1080p indeed. I bought it from aliexpress. It reads this on the pcb: HBVCAM 1718 HD v44

Also see this links

[1 - Amazon: ](https://www.amazon.com/Camera-Module%EF%BC%8CHBV-1716WA-Million-Pixels-Industrial/dp/B07QR5KFP5)

[2 - Vendor](https://www.hbvcamera.com/2-Mega-%20Pixel-usb-cameras/hbvcam-1716-2710-s1.0.html)

// Edit
Best resolution I can get with *guvcview* is 1280*720px which produces the same error above using opencv settings with this resolution

**Comment by JoshChristie** [comment]
Try swapping to V4L backend instead of gstreamer
```
gocv.OpenVideoCaptureWithAPI("/dev/video0", gocv.VideoCaptureV4L2)
```

To see the supported resolutions of the cameras install v4l2-ctl and run:
 `v4l2-ctl --device=/dev/video0 --list-formats-ext`

**Comment by patrickhener** [comment]
```
# command-line-arguments
./mycalib.go:16:14: undefined: gocv.OpenVideoCaptureWithAPI
./mycalib.go:16:46: undefined: gocv.VideoCaptureV4L2
```
Both `OpenVideoCaptureWithAPI` and `gocv.VideoCaptureV4L2` is unknown to my system.

**Comment by JoshChristie** [comment]
Looks like you need to update to the latest release 0.25.0

**Comment by patrickhener** [comment]
Nice now this is working. Thanks
